site/content/blog/2020-02-10-maintaining-consistency-in-codebases-with-go-vet.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Maintaining consistency in codebases with Go Vet
+slug: maintaining-consistency-in-codebases-with-go-vet
+date: 2020-02-10
+categories:
+    - "go"
+author: Jesús Espino
+github: jespino
+community: jesus.espino
+---
+
+At Mattermost we are a bit obsessed with the quality of our code, and one of
+the things that we love is consistent and idiomatic Go. In that sense, go vet
+is a great tool, because it checks a lot of things for us (useless assignments,
+wrong number of Printf parameters, correct format in struct tags, etcetera) as
+part of our CI pipeline, helping us and our contributors to write better code.
+
+The main restriction of go vet is that there are a limited number of absolute
+truths about what is right and what is wrong and so the go vet checks are very
+general.
+
+In Mattermost we have a way of doing certain things like logging, test
+assertions or license headers. We love to keep code consistent so we work hard
+to avoid reintroducing old patterns in the code.
+
+I think the best way to explain it is with an example.
+
+Some time ago we decided to migrate our logging approach to use structured
+logging everywhere, but the migration wasn't complete and the old pattern
+(logging without structured approach) was starting to appear again, in some
+cases mimicking code that wasn’t migrated to the new pattern yet, and in other
+cases by coincidence because the old pattern was one of the obvious approaches
+that you can follow to present data in the logs.
```

Comment:
Some time ago, we re-designed our logging implementation and I think it would be a great example to showcase our work about maintaining consistency. While migrating to the new approach it didn't happen with snapping a finger hence, we observed that the old way of logging was making its way to the new PRs. Also in such cases, the old pattern was one of the obvious approaches that you can follow to present data in the logs.